src/dq_suite/validation.py

```diff
@@ -148,21 +148,86 @@ def _get_gx_expectation_object(
 
         gx_expectation_parameters["meta"] = {
             "table_name": table_name,
-            "column_name": column_name,
-            "expectation_name": gx_expectation_name,
+            "column": column_name,
+            "rule_name": gx_expectation_name,
         }
         return gx_expectation_class(**gx_expectation_parameters)
 
+
+    def _create_geo_expectation(self, expectation_suite_obj, validation_rule):
+        """
+        Create a spatial validation rule for Great Expectations on Spark/Sedona.
+
+        It reads `validation_rule` and builds a SQL filter that returns rows
+        violating the rule (the “unexpected” rows). It also sets a short, human-readable
+        description for reporting.
+        """
+        rule_name = validation_rule.get("rule_name")
+        params = validation_rule.get("parameters", {})
+        geometry_column = params.get("column", "geometry")
+        expected_geometry_type = params.get("geometry_type")
+
+        # Default template
+        base_query_template = """
+            SELECT * FROM {{batch}}
+            WHERE {where_condition}
+        """
+
+        geo_rules = {
+            "ExpectColumnValuesToHaveValidGeometry": (
+                f"NOT ST_IsValid({geometry_column})",
```

Comment:
As mentioned before, I think DBR>=17.1 should include spatial functions such as `ST_IsValid` and `ST_IsEmpty`, see https://docs.databricks.com/aws/en/sql/language-manual/sql-ref-st-geospatial-functions